File: content/blog/maximum_twin_sum_of_a_linked_list/ts/solution_slice.ts
import { ListNode } from './ListNode';

export function pairSumSlice(head: ListNode): number {
    let list: number[] = [head.val];

    // Проходим по списку и заполняем массив значениями
    for (let cur = head.next; cur !== null; cur = cur.next) {
        list.push(cur.val);
    }

    let res = 0;

    // Находим максимальную пару значений
    for (let i = 0; i < Math.floor(list.length / 2); i++) {
        let j = list.length - i - 1;
        res = Math.max(res, list[i] + list[j]);
    }

    return res;
}

File: content/blog/maximum_twin_sum_of_a_linked_list/go/solution_slice.go
package maximum_twin_sum_of_a_linked_list

func pairSumSlice(head *ListNode) int {
	list := []int{head.Val}
	for cur := head.Next; cur != nil; cur = cur.Next {
		list = append(list, cur.Val)
	}

	res := 0
	for i := 0; i < len(list)/2; i++ {
		j := len(list) - i - 1
		res = max(res, list[i]+list[j])
	}

	return res
}
